hummingbot/client/performance.py

```diff
@@ -52,210 +54,236 @@ def __init__(self):
-            sell.price = sum(aggregate_price) / len(aggregate_price)
-            sell.amount = sum(aggregate_amount)
-            aggregated_sells.append(sell)
-
-    return aggregated_buys, aggregated_sells
-
-
-def derivative_pnl(long: list, short: list):
-    # It is assumed that the amount and leverage for both open and close orders are thesame.
-    """
-    Calculates PnL for a close position
-    :param long: a list containing pairs of open and closed long position orders
-    :param short: a list containing pairs of open and closed short position orders
-    :return: A list containing PnL for each closed positions
-    """
-    pnls = []
-    for lg in long:
-        pnls.append((lg[1].price - lg[0].price) * lg[1].amount)
-    for st in short:
-        pnls.append((st[0].price - st[1].price) * st[1].amount)
-    return pnls
-
-
-def is_trade_fill(trade):
-    return type(trade) == TradeFill
-
-
-def are_derivatives(trades: List[Any]) -> bool:
-    return trades and is_trade_fill(trades[0]) and "NILL" not in [t.position for t in trades]
-
-
-async def calculate_performance_metrics(exchange: str,
-                                        trading_pair: str,
-                                        trades: List[Any],
-                                        current_balances: Dict[str, Decimal]) -> PerformanceMetrics:
-    """
-    Calculates PnL, fees, Return % and etc...
-    :param exchange: the exchange or connector name
-    :param trading_pair: the trading market to get performance metrics
-    :param trades: the list of TradeFill or Trade object
-    :param current_balances: current user account balance
-    :return: A PerformanceMetrics object
-    """
-
+    @classmethod
+    async def create(cls, exchange: str,
+                     trading_pair: str,
+                     trades: List[Any],
+                     current_balances: Dict[str, Decimal]) -> 'PerformanceMetrics':
+        performance = PerformanceMetrics()
+        await performance._initialize_metrics(exchange, trading_pair, trades, current_balances)
+        return performance
+
+    @staticmethod
+    def position_order(open: list, close: list) -> Tuple[Any, Any]:
+        """
+        Pair open position order with close position orders
+        :param open: a list of orders that may have an open position order
+        :param close: a list of orders that may have an close position order
+        :return: A tuple containing a pair of an open order with a close position order
+        """
+        result = None
+
+        try:
+            firstWithOpenPosition = next((order for order in open if order.position == "OPEN"))
+            firstWithClosePosition = next((order for order in close if order.position == "CLOSE"))
+            open.remove(firstWithOpenPosition)
+            close.remove(firstWithClosePosition)
+            result = (firstWithOpenPosition, firstWithClosePosition)
```

Comment:
Considering that we are sticking with snake case 🐍  , lets do the same for these variables.